src/CommunityToolkit.Maui/Views/Popup/PopupExtensions.shared.cs

```diff
@@ -43,13 +52,26 @@ public static partial class PopupExtensions
 	/// </returns>
 	public static Task<object?> ShowPopupAsync<TPopup>(this Page page, TPopup popup) where TPopup : Popup
 	{
-#if WINDOWS
-		return PlatformShowPopupAsync(popup, GetMauiContext(page));
-#else
+		if (page.IsPlatformEnabled)
+		{
+			return CreateAndShowPopupAsync(page, popup);
+		}
+		else
+		{
+			var taskCompletionSource = new TaskCompletionSource<object?>();
 
-		CreatePopup(page, popup);
-		return popup.Result;
-#endif
+			void handler(object? sender, NavigatedToEventArgs args)
+			{
+				page.NavigatedTo -= handler;
+
+				CreateAndShowPopupAsync(page, popup)
+					.ContinueWith(antecedent => taskCompletionSource.TrySetResult(antecedent.Result));
```

Comment:
Just a final suggestion and we're good to merge it.
```suggestion
.ContinueWith(antecedent =>
					{
						if (antecedent.Status == TaskStatus.RanToCompletion)
						{
							taskCompletionSource.TrySetResult(antecedent.Result);
						}
						else if (antecedent.Status == TaskStatus.Faulted && antecedent.Exception?.InnerException is Exception ex)
						{
							taskCompletionSource.TrySetException(ex);
						}
					});
```

Reply:
Ha I hadn't considered the exception scenario here... good call